vault/token_store.go

```diff
@@ -1637,8 +1724,13 @@ func (ts *TokenStore) handleLookup(
 		return logical.ErrorResponse("missing token ID"), logical.ErrInvalidRequest
 	}
 
+	lock := ts.getTokenLock(id)
+	lock.RLock()
+	defer lock.RUnlock()
+
 	// Lookup the token
-	out, err := ts.Lookup(id)
+	saltedId := ts.SaltID(id)
+	out, err := ts.lookupSalted(saltedId, true)
```

Comment:
I assume we moved from Lookup to lookupSalted to set `tainted` to `true`. I am curious to know why we need to set tainted while doing a `handleLookup`.